Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -864,4 +866,119 @@ ErrorOr<GC::Ref<Key>> convert_a_value_to_a_multi_entry_key(JS::Realm& realm, JS:
     return convert_a_value_to_a_key(realm, value);
 }
 
+// https://w3c.github.io/IndexedDB/#evaluate-a-key-path-on-a-value
+WebIDL::ExceptionOr<ErrorOr<JS::Value>> evaluate_key_path_on_a_value(JS::Realm& realm, JS::Value value, KeyPath key_path)
+{
+    // 1. If keyPath is a list of strings, then:
+    if (key_path.has<Vector<String>>()) {
+        auto& key_path_list = key_path.get<Vector<String>>();
+
+        // 1. Let result be a new Array object created as if by the expression [].
+        auto result = MUST(JS::Array::create(realm, 0));
+
+        // 2. Let i be 0.
+        u64 i = 0;
+
+        // 3. For each item of keyPath:
+        for (auto const& item : key_path_list) {
+            // 1. Let key be the result of recursively evaluating a key path on a value with item and value.
+            // 2. Assert: key is not an abrupt completion.
+            // 3. If key is failure, abort the overall algorithm and return failure.
+            auto key = TRY(TRY(evaluate_key_path_on_a_value(realm, value, item)));
```

Comment:
This doesn't check that the returned completion is not abrupt. a FIXME on 2 or an explicit step to check seems in order.